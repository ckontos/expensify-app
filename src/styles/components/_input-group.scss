.input-group {
    flex-direction: row;
    display: flex;
    margin-bottom: $l-size;
    
    @media only screen and (max-width: $desktop-breakpoint) {
        flex-direction: column;
        margin-bottom: $m-size;
    }
}

.input-group__item {
    margin: 0 $s-size 0 0;
    
     @media only screen and (max-width: $desktop-breakpoint) {
        margin: 0 0 $s-size 0;
    }
}