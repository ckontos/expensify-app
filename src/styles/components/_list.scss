.list-header {
    background: $off-white;
    border: 1px solid darken($off-white, 7%);
    color: $grey;
    display: flex;
    justify-content: space-between;
    padding: $s-size $m-size;
}

.list-body {
    margin-bottom: $l-size;
    @media (max-width: $desktop-breakpoint) {
        margin-bottom: $m-size;
    }
}

.list-item {
    align-items: center;
    border: 1px solid darken($off-white, 7%);
    border-top: none;
    color: $dark-grey;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding: $m-size;
    text-decoration: none;
    transition: background .3s ease;
    &:hover {
        background: $off-white;
    }
    
    @media (max-width: $desktop-breakpoint) {
        align-items: right;
        display: flex;
        flex-direction: column;
        padding: $s-size;
    }
}

.list-item--message {
    align-items: center;
    justify-content: center;
    color: $grey;
    padding: $m-size;
    &:hover {
        background: none;
    }
}

.list-item__title {
    margin: 0;
    word-break: break-all;
}

.list-item__sub-title {
    color: $grey;
    font-size: $font-size-small;
}

.list-item__data {
    margin: 0;
    padding-left: $s-size;
    
    @media (max-width: $desktop-breakpoint) {
        margin: $s-size 0 0 0;
        padding-left: 0;
    }
}