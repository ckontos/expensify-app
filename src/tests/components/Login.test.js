import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/Login';

test('should render login component', () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});

test('should call the startLogin button', () => {
    const startLogin = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={startLogin} />)
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
});
