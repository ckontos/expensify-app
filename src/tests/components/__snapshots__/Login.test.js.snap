// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login component 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Cash Tracker
    </h1>
    <p>
      It's time to track your spending!
    </p>
    <button
      className="button"
    >
      Login With Google
    </button>
  </div>
</div>
`;
